### ISSUE TITLE
last.ckpt is a soft link of best.ckpt

### ISSUE THREAD

**Issue by Jinbo-Hu** [Lightning-AI/pytorch-lightning#19189]
### Bug description

When I save the last.ckpt, the last.ckpt is symlink of the best.ckpt. I have already set model_checkpoint.save_last=True.

### What version are you seeing the problem on?

v2.1, master

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

cc @carmocca @awaelchli

**Comment by awaelchli** [comment]
The was a new feature in 2.1. You can find some discussion here: #18995
Sorry if it came unexpected.

**Comment by Jinbo-Hu** [comment]
But how do I save the latest or newest epoch? I just upgraded the version from 2.0.4 to 2.1.3.

**Comment by Jinbo-Hu** [comment]
> But how do I save the latest or newest epoch? I just upgraded the version from 2.0.4 to 2.1.3.

automatically

**Comment by Eleven1Liu** [comment]
In my case, when I set `model_checkpoint.save_last=True`, the last checkpoint links to `best_model.ckpt`.
However, `best_model.ckpt` is different from `last.ckpt`. In my understanding, `best_model.ckpt` is the best model you get by validation metric. `last.ckpt` is the model of the last epoch. They are not the same.

The second error we got was when we use 
```
trainer.fit(model, train_loader)
```
without `val_loader` (training for a fixed number of epochs without validation)

We got OSError: [Error 40] Too many levels of symbolic links: '...' because `last.ckpt` is linked to `last.ckpt`.
I think this PR is helpful for us.
